---
title: "DCS Typhi in Blantyre"
author: "Philip Ashton"
date: "`r Sys.Date()`"
output: 
  html_document:
      toc: true
      toc_float: true
      theme: united
      number_sections: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message=FALSE, warning=FALSE)
```

In this document we are analysing Salmonella Typhi from the STRATAA and TyVAC studies. In particular, we are interested in the relationship between QRDR S. Typhi (i.e. Typhi that has mutations in the quinolone resistance determining regions) and the number of ciprofloxacin prescriptions. Other co-variables included in the model are: number of typhoid cases and number of blood cultures carried out.

# Libraries & function imports

```{r message=FALSE}
library(tidyverse)
library(readxl)
library(kableExtra)
#library(lubridate)
#library(here)
#library(tsibble)
#library(feasts)
#library(trending)

source('pefloxacin_typhi_functions.R')
```

# Key variables

`tyvac_pop` is the ndirande intention to treat population (or should we use the per protocol number?). `strataa_pop` is the population size at the final census (aug 2018-April 2019). `tyvac_strataa_start` and `tyvac_strataa_end` are the start and finish of the period of overlap between strataa and tyvac.

```{r KeyVariables}
tyvac_pop <- 17695 # this needs to be updated before it can be used, as is only Ndirande.
strataa_pop <- 102242 # 
cases_start <- as.Date('2016-10-01')
cases_end <- as.Date('2019-10-31')
#tyvac_strataa_start <- as.Date('2018-05-01')
#tyvac_strataa_end <- as.Date('2019-10-01')
#strataa_start <- as.Date('2016-11-01')
#strataa_end <- as.Date('2019-10-31')
```

# Import data

Here we import all the files needed for this analysis.

```{r ImportData, warning=FALSE}

typhis <- get_typhi_data(line_list_handle)
one_typhi_per_patient <- typhis$one_typhi_per_patient
all_typhi <- typhis$all_typhi

strataa_prescriptions_handle <- "/Users/flashton/Dropbox/GordonGroup/STRATAA_pefloxacin_resistant_typhi/data_munging/2021.04.08/2021.04.01 strataa_ndirande_prescribed from James Meiring.csv"
strataa_prescriptions <- read_strataa_prescriptions(strataa_prescriptions_handle)

tyvac_prescrip_detailed <- read_excel("~/Dropbox/GordonGroup/STRATAA_pefloxacin_resistant_typhi/data_munging/2022.09.21/Copy of TyVAC Antibiotic dataset.xlsx", 
    col_types = c("text", "numeric", "numeric", 
        "numeric", "text", "date", "text", 
        "text", "text", "numeric", "text", 
        "numeric", "text", "numeric", "text", 
        "numeric", "text", "numeric", "text", 
        "numeric", "text", "numeric", "text", 
        "numeric", "text", "text", "numeric"))
tyvac_prescrip_detailed$Month <- as.Date(cut(tyvac_prescrip_detailed$`Record Date`, breaks = "month"))

# these data is already tabulated by month
tyvac_presc_by_month <- read_csv("/Users/flashton/Dropbox/GordonGroup/STRATAA_pefloxacin_resistant_typhi/data_munging/2021.04.08/2021.04.08 cipro prescriptions.csv", col_types = cols(Month = col_date(format = "%d/%m/%Y")))
tyvac_blood_cultures_by_month <- read_csv("~/Dropbox/GordonGroup/STRATAA_pefloxacin_resistant_typhi/data_munging/2021.11.12/TyVAC monthly BC.csv", col_types = cols(date = col_date(format = "%Y-%m-%d"))) %>% rename('Month' = 'date', 'tyvac_bc_n' = 'BC collected')

strataa_blood_cultures <- read_csv("~/Dropbox/GordonGroup/STRATAA_pefloxacin_resistant_typhi/data_munging/2021.11.12/STRATAA.ps_enrolment.date_only.csv", col_types = cols(ps12_dov = col_date(format = "%Y-%m-%d")))
strataa_blood_cultures$Month <- as.Date(cut(strataa_blood_cultures$ps12_dov, breaks = "month"))

all_malawian_genomes <- read_excel("/Users/flashton/Dropbox/GordonGroup/STRATAA_pefloxacin_resistant_typhi/phylo/results/2021.11.09/all_malawian/2022.08.30.pefloxacin_typhi_paper_line_list.all_malawian_for_annot.xlsx")

malawi_and_zim_genomes <- read_excel("/Users/flashton/Dropbox/GordonGroup/STRATAA_pefloxacin_resistant_typhi/phylo/results/2021.11.09/malawi_and_zim_clade/prespecified_model_outlier/2021.11.09.MWI_ZIM_clade.with_outlier.annot.xlsx")

malawian_and_context <- read_excel("/Users/flashton/Dropbox/GordonGroup/STRATAA_pefloxacin_resistant_typhi/phylo/results/2021.11.09/malawi_and_context/2022.08.30.pefloxcain_typhi_paper_line_list.malawi_and_context_annot.xlsx")



```

# QECH pef testing results

```{r}
qech_pef_tested_typhi <- one_typhi_per_patient %>% filter(grepl('QECH', Study)) %>% filter(!is.na(`Cipro MIC res?`))
qech_pef_by_month <- qech_pef_tested_typhi %>% group_by(Month, `Cipro MIC res?`) %>% summarise(n = n())
qech_pef_by_month <- qech_pef_by_month %>% filter(Month >= "2019-01-01")

write_csv(qech_pef_by_month, 'data/2022.09.14.qech_pef_results_per_month.csv')
```

# Calculate the total number of microbiologically confirmed typhi in STRATAA/TyVAC

```{r}
# opp is one per patient
micro_confirmed_opp <- get_strataa_tyvac_cases(one_typhi_per_patient, cases_start, cases_end) 
micro_confirmed_opp_by_month <- micro_confirmed_opp %>% group_by(Month) %>% summarise(micro_confirmed_opp_n = n())
# all includes mutliple samples from single patients
micro_confirmed_all <- get_strataa_tyvac_cases(all_typhi, cases_start, cases_end)
micro_confirmed_all_by_month <- micro_confirmed_all %>% group_by(Month) %>% summarise(micro_confirmed_all_n = n())
```

# Sequencing confirmed typhoid incidence rate

Here we filter the input line list to extract the *sequenced* tyvac samples from ndirande and the sequenced strataa samples.

```{r}

sequencing_confirmed_typhi <- one_typhi_per_patient %>% filter(!is.na(`Sequenced?`) & `overall sequencing identification` == "Typhi")

#tyvac_cases <- sequencing_confirmed_typhi %>% filter(Study == 'TyVAC' & `TyVAC site` == 'Ndirande')
tyvac_cases <- sequencing_confirmed_typhi %>% filter(Study == 'TyVAC')
tyvac_cases_by_month <- tyvac_cases %>% group_by(Month) %>% summarise(tyvac_n = n())

# we use grepl to find "strataa adjacent" samples as well as strataa samples.
strataa_cases <- sequencing_confirmed_typhi %>% filter(grepl('STRATAA', Study))
strataa_cases_by_month <- strataa_cases  %>% group_by(Month) %>% summarise(strataa_n = n())
strataa_cases_by_month$strataa_typhoid_per100000 <- (strataa_cases_by_month$strataa_n / strataa_pop) * 100000

combined_cases_by_month <- combine_by_month(tyvac_cases_by_month, strataa_cases_by_month, cases_start, cases_end, 'cases')

```

# QRDR Typhi incidence

```{r}

strataa_qrdr <- strataa_cases %>% filter(any_qrdr == 'QRDR')
strataa_qrdr_cases_by_month <- strataa_qrdr  %>% group_by(Month) %>% summarise(strataa_qrdr_n = n())
strataa_qrdr_cases_by_month$strataa_qrdr_incidence_per_100000 <- (strataa_qrdr_cases_by_month$strataa_qrdr_n / strataa_pop) * 100000

tyvac_qrdr <- tyvac_cases %>% filter(any_qrdr == 'QRDR')
tyvac_qrdr_cases_by_month <- tyvac_qrdr  %>% group_by(Month) %>% summarise(tyvac_qrdr_n = n())


combined_qrdr_cases_by_month <- combine_by_month(tyvac_qrdr_cases_by_month, strataa_qrdr_cases_by_month, cases_start, cases_end, 'qrdr')


```

# Cipro prescription incidence rates

STRATAA

```{r}
strataa_presc_by_month <- strataa_prescriptions %>% group_by(Month) %>% summarise(strataa_prescrip_n = n())
strataa_presc_by_month$strataa_prescrip_per100000 <- (strataa_presc_by_month$strataa_prescrip_n / strataa_pop) * 100000

combined_prescrip_by_month <- combine_by_month(tyvac_presc_by_month, strataa_presc_by_month, cases_start, cases_end, 'prescrip')

```

TyVAC from detailed

```{r}
tyvac_prescrip_detailed_cipro <- tyvac_prescrip_detailed %>% filter(`1i. Ciprofloxacin` == 'Yes') %>% select(`residence`, `Record Date`, `1i. Ciprofloxacin`, Month) 
tyvac_prescrip_detailed_cipro_by_month <- tyvac_prescrip_detailed_cipro %>% group_by(Month, residence) %>% summarise(n = n())

tyvac_prescrip_detailed_cipro_by_month_w <- tyvac_prescrip_detailed_cipro_by_month %>% pivot_wider(values_from = n, names_from = residence)
tyvac_prescrip_detailed_cipro_by_month_w[is.na(tyvac_prescrip_detailed_cipro_by_month_w)] <- 0 

tyvac_prescrip_detailed_cipro_by_month_w <- tyvac_prescrip_detailed_cipro_by_month_w %>% mutate(total = Zingwangwa + Ndirande + `NA`)

ggplot(tyvac_prescrip_detailed_cipro_by_month, aes(x = Month, y = n, fill = residence)) + geom_bar(stat = "identity")

```

# Getting the number of blood cultures

```{r}

strataa_blood_cultures_by_month <- strataa_blood_cultures %>% group_by(Month) %>% summarise(strataa_blood_cultures_n = n())
strataa_blood_cultures_by_month$strataa_bc_per_100000 <- (strataa_blood_cultures_by_month$strataa_blood_cultures_n / strataa_pop) * 100000

combined_blood_cultures_by_month <- combine_by_month(tyvac_blood_cultures_by_month, strataa_blood_cultures_by_month, cases_start, cases_end, 'blood_cultures')

```

# Make a clean dataframe for analysis

```{r}

entire_study_period <- combine_cases_prescrip_qrdr_bc_mc(combined_qrdr_cases_by_month, combined_cases_by_month, combined_prescrip_by_month, combined_blood_cultures_by_month, micro_confirmed_opp_by_month, micro_confirmed_all_by_month, cases_start, cases_end)

entire_study_period[is.na(entire_study_period)] <- 0 

entire_study_period  %>% kbl() %>% kable_styling()


write_csv(entire_study_period, 'data/2022.09.14.pef_typhi.cases_bc_qrdr_prescrip_microconf.csv')
```


# Get the occurrence of the different qrdr mutations

Filter to get only samples that have a QRDR mutation, that were either in tyvac-ndirande or strataa.

```{r}
#strataa_tyvac_qrdr_typhi <- sequencing_confirmed_typhi %>% filter(!is.na(QRDR)) %>% filter((Study == 'TyVAC' & `TyVAC site` == 'Ndirande') | grepl('STRATAA', Study)) %>% arrange(`Properly formatted date`)
strataa_tyvac_qrdr_typhi <- sequencing_confirmed_typhi %>% filter(!is.na(QRDR)) %>% filter(Study == 'TyVAC' | grepl('STRATAA', Study)) %>% arrange(`Properly formatted date`)

write_csv(strataa_tyvac_qrdr_typhi, 'data/2022.09.15.strataa_tyvac_qrdr_typhi.one_per_patient.csv')

```

# Get the total number of sequenced S. Typhi included in each phylogeny

First, the number of sequenced genomes from this study (will change after strataa paper published) and context from Malawi
```{r}

malawian_genomes_summary <- all_malawian_genomes %>% 
  mutate(study_summary = NA) %>%
  mutate(study_summary = ifelse(Study %in% list('wong et al., 2015', 'Mashe_et_al_2021'), 'context', study_summary)) %>% 
  mutate(study_summary = ifelse(Study %in% list('STRATAA adjacent', 'QECH', 'TyVAC', 'STRATAA'), 'this_study', study_summary))  %>%
  filter(!grepl('Ignore', note)) %>% 
  group_by(Country, study_summary) %>% summarise(n = n()) %>% mutate(tree = 'all_malawian')


malawi_zim_genomes_summary <- malawi_and_zim_genomes %>% 
  mutate(study_summary = NA) %>%
  mutate(study_summary = ifelse(Study %in% list('wong et al., 2015', 'Mashe_et_al_2021'), 'context', study_summary)) %>% 
  mutate(study_summary = ifelse(Study %in% list('STRATAA adjacent', 'QECH', 'TyVAC', 'STRATAA'), 'this_study', study_summary))  %>%    
  filter(!grepl('Ignore', note)) %>% 
  group_by(Country, study_summary) %>% summarise(n = n()) %>% mutate(tree = 'malawi_and_zim')

malawian_and_context_summary <- malawian_and_context %>% mutate(study_summary = NA) %>%
  mutate(study_summary = ifelse(Study %in% list('wong et al., 2015', 'Mashe_et_al_2021', 'park et al., 2018'), 'context', study_summary)) %>% 
  mutate(study_summary = ifelse(Study %in% list('STRATAA adjacent', 'QECH', 'TyVAC', 'STRATAA'), 'this_study', study_summary))  %>%  
  filter(!grepl('Ignore', Note)) %>% 
  group_by(`Geog for itol`, study_summary) %>% summarise(n = n()) %>% 
  rename(Country = `Geog for itol`) %>% mutate(tree = 'malawian_and_context')

all_tree_summaries <- do.call("rbind", list(malawian_genomes_summary, malawi_zim_genomes_summary, malawian_and_context_summary))

write_csv(all_tree_summaries, 'data/2022.09.16.tree_summaries.csv')

```




